test/edu/ithaca/dragonlab/ckc/conceptgraph/ConceptGraphTest.java

```diff
@@ -1,30 +1,222 @@
+        Hashtable<String, Integer> actualCase = suggestedConceptNodeTest(objectSuggestionMap);
+
+        Assert.assertEquals(actualCase,testCase);
+    }
+
+
+    @Test
+    public void SuggestedConceptNodeMapWillOneStudentTest() {
+        ConceptGraph orig= PracticalConceptGraphTest.willExampleConceptGraphTestOneStudent();
+        //what it is
+        HashMap<ConceptNode, List<learningObjectSuggestion>> objectSuggestionMap = orig.SuggestedConceptNodeMap();
+        //what it should be
+        HashMap<ConceptNode, List<learningObjectSuggestion>> mapTest = new HashMap<>();
+
+        List<learningObjectSuggestion> testList = new ArrayList<>();
+        testList.add(new learningObjectSuggestion("Q13",1, learningObjectSuggestion.Level.INCOMPLETE));
+        mapTest.put(new ConceptNode("Counting"), testList);
+
+        List<learningObjectSuggestion> testList2 = new ArrayList<>();
+        testList2.add(new learningObjectSuggestion("Q10", 1, learningObjectSuggestion.Level.INCOMPLETE));
+        testList2.add(new learningObjectSuggestion("Q6", 1, learningObjectSuggestion.Level.INCOMPLETE));
+        mapTest.put(new ConceptNode("Boolean"), testList2);
+
+        Hashtable<String, Integer> actualCase = suggestedConceptNodeTest(objectSuggestionMap);
+        Hashtable<String, Integer> testCase = suggestedConceptNodeTest(mapTest);
+
+        Assert.assertEquals(actualCase,testCase);
+    }
+
+
+    @Test
+    public void buildLearningObjectListSimpleTest(){
+        ConceptGraph orig = ExampleConceptGraphFactory.makeSimple();
+        orig.addLearningObjects(ExampleLearningObjectFactory.makeSimpleLearningObjectDef());
+
+        //from A nodes
+        HashMap<String, Integer> testCompareA = new HashMap<String, Integer>();
+        testCompareA.put("Q1",1);
+        testCompareA.put("Q2",1);
+        testCompareA.put("Q3",2);
+        testCompareA.put("Q4",2);
+        testCompareA.put("Q5",2);
+        testCompareA.put("Q6",2);
+
+        Assert.assertEquals(testCompareA, orig.buildLearningObjectSummaryList("A"));
+
+        //from B node
+        HashMap<String, Integer> testCompareB = new HashMap<String, Integer>();
+        testCompareB.put("Q1",1);
+        testCompareB.put("Q2",1);
+        testCompareB.put("Q3",1);
+        testCompareB.put("Q4",1);
+        testCompareB.put("Q5",1);
+        testCompareB.put("Q6",1);
+
+        Assert.assertEquals(testCompareB, orig.buildLearningObjectSummaryList("B"));
+
+
+        //from c node
+        HashMap<String, Integer> testCompareC = new HashMap<String, Integer>();
+        testCompareC.put("Q3",1);
+        testCompareC.put("Q4",1);
+        testCompareC.put("Q5",1);
+        testCompareC.put("Q6",1);
+
+        Assert.assertEquals(testCompareC, orig.buildLearningObjectSummaryList("C"));
+
+        //from a node not in graph
+        Assert.assertEquals(null, orig.buildLearningObjectSummaryList("W"));
+    }
+
+
+    @Test
+    public void suggestionComparatorTest(){
```

Comment:
This should be in a separate test class, either SuggestionComparatorTest or LearningObjectSuggestionTest